src/github.com/matrix-org/dendrite/clientapi/routing/login.go

```diff
@@ -37,10 +46,15 @@ type flow struct {
 	Stages []string `json:"stages"`
 }
 
-type passwordRequest struct {
-	User               string  `json:"user"`
-	Password           string  `json:"password"`
-	InitialDisplayName *string `json:"initial_device_display_name"`
+type loginRequest struct {
```

Comment:
I would put this type in the [authtypes](https://github.com/matrix-org/dendrite/tree/master/src/github.com/matrix-org/dendrite/clientapi/auth/authtypes) folder as well, in a new file.